src/java/org/apache/cassandra/index/sai/disk/v2/V2VectorIndexSearcher.java

```diff
@@ -195,21 +203,36 @@ private BitsOrPostingList bitsOrPostingListForKeyRange(QueryContext context, Abs
             // If num of matches are not bigger than limit, skip ANN.
             // (nRows should not include shadowed rows, but context doesn't break those out by segment,
             // so we will live with the inaccuracy.)
-            var nRows = Math.toIntExact(maxSSTableRowId - minSSTableRowId + 1);
+            int nRows = Math.toIntExact(maxSSTableRowId - minSSTableRowId + 1);
             int maxBruteForceRows = min(globalBruteForceRows, maxBruteForceRows(limit, nRows, graph.size()));
-            logger.trace("Search range covers {} rows; max brute force rows is {} for sstable index with {} nodes, LIMIT {}",
+            if (logger.isTraceEnabled())
```

Comment:
this isn't necessary since none of the parameters provided to the method are expensive

Reply:
parameters are ints, so the trace() will box them into Integer. this check saves us from autoboxing.

Reply:
good point, thanks